sql/core/src/test/scala/org/apache/spark/sql/execution/PlannerSuite.scala

```diff
@@ -32,6 +32,26 @@ import org.apache.spark.sql.types._
 
 
 class PlannerSuite extends FunSuite {
+  test("multiway full outer join") {
+    val planned = testData
+                  .join(testData2, testData("key") === testData2("a"), "outer")
+                  .join(testData3, testData("key") === testData3("a"), "outer")
+                  .queryExecution.executedPlan
+    val exchanges = planned.collect { case n: Exchange => n }
+
+    assert(exchanges.size === 3)
```

Comment:
This requires some further change I think, @yhuai should have some idea on this.